opentracing-api/src/main/java/io/opentracing/ThreadLocalScheduler.java

```diff
@@ -0,0 +1,70 @@
+package io.opentracing;
+
+/**
+ * ThreadLocalScheduler is a trivial Scheduler implementation that relies on Java's thread-local storage primitives.
+ *
+ * @see Scheduler
+ * @see Tracer#scheduler()
+ */
+public class ThreadLocalScheduler implements Scheduler {
+    ThreadLocal<Continuation> threadLocalActive = new ThreadLocal<>();
+
+    @Override
+    public Span active() {
+        Continuation state = threadLocalActive.get();
+        return (state == null) ? null : state.span;
+    }
+
+    @Override
+    public Continuation capture(Span span) {
+        return new Continuation(span);
+    }
+
+    @Override
+    public SpanContext activeContext() {
+        Span active = this.active();
+        if (active == null) return null;
+        return active.context();
+    }
+
+    @Override
+    public Continuation captureActive() {
+        return capture(active());
+    }
+
+    class Continuation implements Scheduler.Continuation {
+        private final Span span;
+        private boolean finishOnDeactivate;
+        private Continuation toRestore = null;
+
+        @Override
+        public Span activate(boolean finishOnDeactivate) {
+            this.finishOnDeactivate = finishOnDeactivate;
+            toRestore = threadLocalActive.get();
+            threadLocalActive.set(this);
+            return null;
+        }
+
+        @Override
+        public void close() {
+            this.deactivate();
+        }
+
+        @Override
+        public void deactivate() {
+            if (span != null && this.finishOnDeactivate) {
+                span.finish();
+            }
```

Comment:
I assume this is meant to be used with auto-close via `try(continuation)`. Then the following is also possible
```java
try (continuation) {
    try(Span span = continuation.activate()) {
    }
}
```
which technically removes the need for auto-finish flag. But it doesn't look like we can use that due to java-6 req.